policy/compiler/compiler.go

```diff
@@ -335,6 +335,41 @@ func (tc *templateCompiler) compileOpenAPISchema(dyn *model.DynValue,
 	if found {
 		schema.DefaultValue = tc.convertToSchemaType(elem.Ref.ID, elem.Ref.Value, schema)
 	}
+
+	tc.validateOpenAPISchema(dyn, schema)
+}
+
+func (tc *templateCompiler) validateOpenAPISchema(dyn *model.DynValue, schema *model.OpenAPISchema) {
+	m := tc.mapValue(dyn)
+	if len(m.Fields) > 0 && len(schema.Type) == 0 {
```

Comment:
I think this check will be redundant with the one on line 352 since the m.Fields corresponds to the presence of either `properties` or `additionalProperties`

Reply:
removed